lib/collection/src/operations/verification/mod.rs

```diff
@@ -0,0 +1,160 @@
+mod search;
+
+use std::fmt::Display;
+
+use segment::types::Filter;
+
+use super::config_diff::StrictModeConfig;
+use crate::collection::Collection;
+
+/// Trait to verify strict mode for requests. All functions in this trait will default to an 'empty' implementation,
```

Comment:
I don't like this idea too much. It makes it more ergonomic to implement, but it may shoot us in the foot when maintaining it. 

IMO it is better to explicitly ignore irrelevant methods for each implementor and let the compiler complain if something isn't considered.